packages/ui/stories/Integration/IntegrationDetail.stories.tsx

```diff
@@ -0,0 +1,238 @@
+import {
+  Button,
+  DropdownKebab,
+  Grid,
+  ListViewItem,
+  MenuItem,
+} from 'patternfly-react';
+import { text } from '@storybook/addon-knobs';
+import { withNotes } from '@storybook/addon-notes';
+import { storiesOf } from '@storybook/react';
+import * as React from 'react';
+
+import { StoryHelper } from '../../.storybook/StoryHelper';
+import { IntegrationDetail } from '../../src';
+
+const stories = storiesOf('Integration/IntegrationDetail', module);
+const storyNotes = 'Integration Detail';
+
+const integrationId = 'i-LUF4Pfwxo4Wcrbyt7YIz';
+const integrationDescription = 'A test integration.';
+const integrationExternalUrl =
+  'https://i-swagger-connections-syndesis-staging.b6ff.rh-idev.openshiftapps.com';
+const integrationName = 'Student API';
+const integrationStatus = 'Published';
+const integrationVersion = '1';
+
+const textBtnEdit = 'Edit';
+const textBtnPublish = 'Publish';
+const textCopyToClipboard = 'Copy to Clipboard';
+const textDraft = 'Draft';
+const textExternalUrl = 'External URL';
+const textHistory = 'History';
+const textHistoryMenuReplaceDraft = 'Replace Draft';
+const textHistoryMenuUnpublish = 'Unpublish';
+const textLastPublished = 'Last published';
+const textNoDescription = 'No description set...';
+const textTabActivity = 'Activity';
+const textTabDetails = 'Details';
+const textTabMetrics = 'Metrics';
+const textTitle = 'Integration Detail';
+const textVersion = 'Version';
+
+const historyItems = [
+  <Grid fluid={true} key={1}>
+    <Grid.Row className="show-grid">
+      <Grid.Col xs={2} md={2}>
+        {<span>{text('i18nTextDraft', textDraft)}:</span>}
+      </Grid.Col>
+      <Grid.Col xs={10} md={10}>
+        <ListViewItem
+          key={1}
+          heading={<i>{text('i18nTextDraft', textDraft)}</i>}
+          actions={
+            <div>
+              <Button>Action 1</Button>
+              <DropdownKebab id="action2kebab" pullRight={true}>
+                <MenuItem>
+                  {text(
+                    'i18nTextHistoryMenuReplaceDraft',
+                    textHistoryMenuReplaceDraft
+                  )}
+                </MenuItem>
+                <MenuItem>
+                  {text(
+                    'i18nTextHistoryMenuUnpublish',
+                    textHistoryMenuUnpublish
+                  )}
+                </MenuItem>
+              </DropdownKebab>
+            </div>
+          }
+          stacked={false}
+          i18nTextBtnEdit={text('', textBtnEdit)}
+          i18nTextBtnPublish={text('', textBtnPublish)}
+          i18nTextHistoryMenuReplaceDraft={text(
+            '',
+            textHistoryMenuReplaceDraft
+          )}
+          i18nTextHistoryMenuUnpublish={text('', textHistoryMenuUnpublish)}
+          i18nTextLastPublished={text('', textLastPublished)}
+        />
+      </Grid.Col>
+    </Grid.Row>
```

Comment:
Can we refactor this - quite complicated - code in specialized components? We don't want to ask to the consumer of the IntegrationDetail to know this much of the inner bits of the UI library.

Reply:
Right, wasn't sure if all of the scenarios (published, unpublished, etc) should all be in the same story and component. I can break it up but just wasn't sure how much (e.g. per tab, also the history section).